lib/model/content.dart

```diff
@@ -1448,6 +1448,43 @@ class _ZulipContentParser {
     return tableNode ?? UnimplementedBlockContentNode(htmlNode: tableElement);
   }
 
+  List<BlockContentNode> parseMathBlocks(dom.Element element) {
+    assert(_debugParserContext == _ParserContext.block);
+    assert(element.localName == 'p');
+    assert((() {
+      final first = element.nodes.first;
+      return first is dom.Element
+        && first.localName == 'span'
+        && first.className == 'katex-display';
+    })());
+    final debugHtmlNode = kDebugMode ? element : null;
+
+    final blocks = <BlockContentNode>[];
+    for (final child in element.nodes) {
+      // The case with the `<br>\n` can happen when at the end of a quote;
+      // it seems like a glitch in the server's Markdown processing,
+      // so hopefully there just aren't any further such glitches.
+      if (child case dom.Element(localName: 'br')) continue;
+      if (child case dom.Text(text: '\n' || '\n\n')) continue;
```

Comment:
Looks like this no longer checks if the `<br>\n` is at the end. Would it be unexpected to have them appear in the middle?

`\n\n` on the other hand, appear to be somehow separate from the quirk commented here.

Reply:
Separated the handling of both these cases.